File: ios/GotchuReviews/GotchuReviews/Models/Review.swift
import Foundation

struct Review: Codable, Identifiable {
    let id: String
    let contractorId: String
    let userId: String
    let ratingQuality: Int
    let ratingCommunication: Int
    let ratingTimeliness: Int
    let ratingValue: Int
    let overallRating: Double
    let title: String
    let body: String
    let workType: String
    let isVerified: Bool
    let invoiceId: String?
    let createdAt: String
    let invoice: InvoiceSummary?
}

struct InvoiceSummary: Codable {
    let totalAmount: Double
    let laborCost: Double?
    let materialsCost: Double?
    let description: String
    let invoiceDate: String?
    let source: String
}

File: android/GotchuReviews/app/src/main/java/com/gotchureviews/app/data/model/Review.kt
package com.gotchureviews.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Review(
    val id: String,
    val contractorId: String,
    val userId: String,
    val ratingQuality: Int,
    val ratingCommunication: Int,
    val ratingTimeliness: Int,
    val ratingValue: Int,
    val overallRating: Double,
    val title: String,
    val body: String,
    val workType: String,
    val isVerified: Boolean,
    val invoiceId: String? = null,
    val createdAt: String,
    val invoice: InvoiceSummary? = null,
)

@Serializable
data class InvoiceSummary(
    val totalAmount: Double,
    val laborCost: Double? = null,
    val materialsCost: Double? = null,
    val description: String,
    val invoiceDate: String? = null,
    val source: String,
)

@Serializable
data class ReviewBrief(
    val id: String,
    val overallRating: Double,
    val title: String,
)
